### ISSUE TITLE
System sending virtual card receipt notifications when receipt already uploaded

### ISSUE THREAD

**Issue by alanna** [opencollective/opencollective#5951]
**Describe the bug**
I uploaded the receipt for this while it was still processing: https://opencollective.com/comms-docs/expenses/93956

Then about 12 hours later, I got this notification:

![Screen Shot 2022-09-13 at 8 53 20 AM](https://user-images.githubusercontent.com/5498878/189756605-cd901255-560b-4594-81cf-e104e65a6ddb.png)

**To Reproduce**
1. Upload a receipt to a VC charge before the email goes out
2. Receive email anyway

**Expected behavior**
The system should not tell you to upload a receipt that is already uploaded

  P3 high frequency, low impact

**Comment by alanna** [comment]
@kewitz this is the correct expense and I definitely uploaded the receipt

**Comment by alanna** [comment]
This happened again @kewitz @SudharakaP https://opencollective.com/foundation/expenses/100328

I uploaded the receipt when it was processing on oct 7, then got a system notification asking for the receipt once it was incurred on oct 9. The system should not request receipts if they have already been uploaded.
<img width="1047" alt="Screen Shot 2022-10-09 at 10 32 00 AM" src="https://user-images.githubusercontent.com/5498878/194728488-914fe156-bbb5-455b-bb18-c806a70d812d.png">
<img width="812" alt="Screen Shot 2022-10-09 at 10 32 20 AM" src="https://user-images.githubusercontent.com/5498878/194728498-daf03a33-6527-47a4-b54b-1eacd743b4c4.png">

**Comment by znarf** [comment]
@SudharakaP can you check that it's working as expected and understand why it didn't work for @alanna ?

**Comment by SudharakaP** [comment]
> @SudharakaP can you check that it's working as expected and understand why it didn't work for @alanna ?

@znarf : Sorry, I missed something in my originator PR. Looking at this further I realized that we also store attachments in `ExpenseItems` table. For example if it's a reimbursement with several `ExpenseItems` it can be the case where we have nothing in `ExpenseAttachedFiles` but everything stored in `ExpenseItems`. So basically we need to take both into account. I've pushed another PR to fix this; https://github.com/opencollective/opencollective-api/pull/8060.
